Sliders are connected to node initial-values.  [#102293636]

knowuh opened the pull request:
Also part of https://www.pivotaltracker.com/story/show/102293636.  This time we have connected the sliders in the graph window to the nodes initial-values.  

There is more work to be done, but I thought I would put this up to talk about.  At the time of this PR, I also deployed to http://concord-consortium.github.io/building-models/

![initial-value-sliders](https://cloud.githubusercontent.com/assets/22908/10178345/2a066e78-66cd-11e5-8002-9721924fcc5b.png)

Changes:  

* Sliders only show up for independent variables, or accumulators.
* Sliders adjust the 'initial value' of nodes.
* The Value text-edit field is the 'current-value' -- which is usually just the initial value.

sfentress commented:
This all looks great to me. The only thing that I could imagine being a part of the PR is being able to set that the Slider's values are all integers. (Or even be able to define that it has n-steps.) It's a pretty common slider functionality, and we'll probably need it soon anyway. When setting the values using the slider, it's pretty much impossible to set them to integers, which feels unexpected, especially as the view tells us it's an integer.

![image](https://cloud.githubusercontent.com/assets/35721/10180185/0c5155ee-66d9-11e5-9113-c12590cb8841.png)

sfentress commented:
This all looks great to me. The only thing that I could imagine being a part of the PR is being able to set that the Slider's values are all integers. (Or even be able to define that it has n-steps.) It's a pretty common slider functionality, and we'll probably need it soon anyway. When setting the values using the slider, it's pretty much impossible to set them to integers, which feels unexpected, especially as the view tells us it's an integer.

<img width="948" alt="screenshot 2015-09-29 18 36 27" src="https://cloud.githubusercontent.com/assets/35721/10180193/258abca8-66d9-11e5-8c91-25a1fb4814f3.png">

knowuh commented:
That is a good point about the sliders.  I like the idea of being able to set the increment of the slider.  I think I will add that to this PR.

knowuh commented:
@sfentress I added increment values to the sliders in https://github.com/concord-consortium/building-models/commit/d0f1042459f4f3c2d1968df7186b248599fa7d90

I also changed the variable layout again in https://github.com/concord-consortium/building-models/commit/596a91fa5884e3a1e79304670715508164b89a86

Please check it out if you have time.

sfentress commented:
The slider changes look great.

Is it just me, or do the new arrows look really far away from the nodes? If that's the design, though, then that's totally fine and this all looks good.

Old network:

<img width="440" alt="screenshot 2015-09-30 22 21 32" src="https://cloud.githubusercontent.com/assets/35721/10211505/cefbbfba-67c1-11e5-9ef6-2e826e6523c2.png">

New network:

<img width="428" alt="screenshot 2015-09-30 22 21 40" src="https://cloud.githubusercontent.com/assets/35721/10211510/d4453848-67c1-11e5-8f0b-74d2cd186272.png">

It's just confusing because in your top image above, the edge between the egg and chick seems to connect directly to the egg, where as here it's far away, so I don't know if there's different states it can be in that might cause that.

Besides that question, though, :+1:

knowuh commented:
Yeah, there is some trade-off that we can discuss in the meeting tomorrow WRT where the arrows display.  Basically they connect to the outside edge of a DIV.  In this case I decided to make an invisible div which encompassed the tittle and the image. Previously it was just the image.  There are also different states, so when there are sliders and other things, I change the div that the arrows point at...  TBD, but I am going to merge.

sfentress commented:
Totally agree.
